src/persist/src/indexed/cache.rs

```diff
@@ -327,7 +320,8 @@ impl<B: Blob> BlobCache<B> {
         let async_guard = self.async_runtime.enter();
 
         let delete_start = Instant::now();
-        self.unsealed.lock()?.remove(&batch.key);
+        self.cache
+            .remove_unsealed(&batch.key, usize::cast_from(batch.size_bytes))?;
```

Comment:
I prefer for the cache to do its own accounting here (save the byte count that was handed in the maybe_add call) instead of relying on this one matching

Reply:
done